public/app/features/datasources/components/picker/utils.test.ts

```diff
@@ -28,3 +28,47 @@ describe('isDataSourceMatch', () => {
     expect(isDataSourceMatch(dataSourceInstanceSettings, { uid: 'b' } as DataSourceRef)).toBeFalsy();
   });
 });
+
+describe('getDataSouceCompareFn', () => {
```

Comment:
Maybe, I'd add a use case to make sure all of them work together. Something like: 
```js
  it('should sort alphabetically ordering the groups like: current > Variables > Rest > Built-in', ...);
```

So we can guarantee that there are two variables or that the built-in DS is also sorted properly. But also that the groups are placed in order as well.

In summary, I think we can simplify the unit tests into one 😄